arch/xtensa/src/esp32/esp32_himem.c

```diff
@@ -320,7 +320,7 @@ static bool allocate_blocks(int count, uint16_t *blocks_out)
       for (i = 0; i < count; i++)
         {
           g_ram_descriptor[blocks_out[i]].is_alloced = true;
-          g_ram_descriptor[blocks_out[i]].is_mapped  = false;
+          assert(g_ram_descriptor[blocks_out[i]].is_mapped  == false);
```

Comment:
should we change to DEBUGASSERT?

Reply:
assert is defined to DEBUGASSERT when CONFIG_DEBUG_ASSERTIONS is defined, otherwise it is defined to do nothing, please see include/assert.h for reference.

Reply:
Ok, I decided to replace assert with DEBUGASSERT to follow the standard.